### PR TITLE
[ML] Track the number of inference failures

### PR DESCRIPTION
**Pull request by davidkyle**
The number of successful inferences is counted but not the number of failures.
The most likely cause of a failure is where the input tensor is not properly constructed for the model

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetDeploymentStatsAction.java [hunk 1]
```diff
@@ -302,6 +302,7 @@ protected void taskOperation(
                     // avoid reporting the average time as 0 if count < 1
                     (stats.get().getTimingStats().getCount() > 0) ? stats.get().getTimingStats().getAverage() : null,
                     stats.get().getPendingCount(),
+                    stats.get().getErrorCount() > 0 ? stats.get().getErrorCount() : null,
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetDeploymentStatsAction.java:305]: If there are no errors setting `null` here means the field will not be in the response

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchResultProcessor.java [hunk 2]
```diff
@@ -93,25 +96,24 @@ public void process(NativePyTorchProcess process) {
...
+            lastUsed
+        );
     }
 
     private synchronized void processResult(PyTorchResult result) {
         if (result.isError() == false) {
             timingStats.accept(result.getTimeMs());
             lastUsed = Instant.now();
+        } else {
+            errorCount++;
```
Comment (benwtrent) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchResultProcessor.java:113]: Do we want to count the following as errors as we well?

 - Queue rejection
 - Timeout
 - missing field?

Comment (davidkyle) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchResultProcessor.java:113]: Yes!

Comment (benwtrent) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchResultProcessor.java:113]: OK, then the error count cannot only be in the process result manager :(. Might complicate things.

**Pull request closed without merging**